### ISSUE TITLE
Error: ECONNREFUSED connect ECONNREFUSED after Chrome version update

### ISSUE THREAD

**Issue by mantisbayne** [angular/protractor#3655]
My protractor e2e tests have suddenly stopped working.  I'm getting the error:

```
[15:44:53] I/testLogger - [chrome ANY ANY #01-12] PID: 58628
[chrome ANY ANY #01-12] Specs: /Users/user/Projects/project/src/e2e/tests/test.spec.js
[chrome ANY ANY #01-12]
[chrome ANY ANY #01-12] [15:44:48] I/direct - Using ChromeDriver directly...
[chrome ANY ANY #01-12]
[chrome ANY ANY #01-12] /Users/user/Projects/project/node_modules/selenium-webdriver/http/index.js:365
[chrome ANY ANY #01-12]       onError(new Error(message));
[chrome ANY ANY #01-12]               ^
[chrome ANY ANY #01-12] Error: ECONNREFUSED connect ECONNREFUSED 127.0.0.1:60998
[chrome ANY ANY #01-12]     at ClientRequest.<anonymous> (/Users/user/Projects/project/node_modules/selenium-webdriver/http/index.js:365:15)
[chrome ANY ANY #01-12]     at emitOne (events.js:90:13)
[chrome ANY ANY #01-12]     at ClientRequest.emit (events.js:182:7)
[chrome ANY ANY #01-12]     at Socket.socketErrorListener (_http_client.js:306:9)
[chrome ANY ANY #01-12]     at emitOne (events.js:90:13)
[chrome ANY ANY #01-12]     at Socket.emit (events.js:182:7)
[chrome ANY ANY #01-12]     at emitErrorNT (net.js:1249:8)
[chrome ANY ANY #01-12]     at _combinedTickCallback (internal/process/next_tick.js:74:11)
[chrome ANY ANY #01-12]     at process._tickCallback (internal/process/next_tick.js:98:9)
[chrome ANY ANY #01-12] From: Task: WebDriver.createSession()
[chrome ANY ANY #01-12]     at acquireSession (/Users/user/Projects/project/node_modules/selenium-webdriver/lib/webdriver.js:62:22)
[chrome ANY ANY #01-12]     at Function.createSession (/Users/user/Projects/project/node_modules/selenium-webdriver/lib/webdriver.js:295:12)
[chrome ANY ANY #01-12]     at Driver (/Users/user/Projects/project/node_modules/selenium-webdriver/chrome.js:778:38)
[chrome ANY ANY #01-12]     at Direct.getNewDriver (/Users/user/Projects/project/node_modules/protractor/built/driverProviders/direct.js:65:26)
[chrome ANY ANY #01-12]     at Runner.createBrowser (/Users/user/Projects/project/node_modules/protractor/built/runner.js:182:43)
[chrome ANY ANY #01-12]     at /Users/user/Projects/project/node_modules/protractor/built/runner.js:255:30
[chrome ANY ANY #01-12]     at _fulfilled (/Users/user/Projects/project/node_modules/q/q.js:834:54)
[chrome ANY ANY #01-12]     at self.promiseDispatch.done (/Users/user/Projects/project/node_modules/q/q.js:863:30)
[chrome ANY ANY #01-12]     at Promise.promise.promiseDispatch (/Users/user/Projects/project/node_modules/q/q.js:796:13)
[chrome ANY ANY #01-12]     at /Users/user/Projects/project/node_modules/q/q.js:556:49

[15:44:53] I/testLogger -

[15:44:53] E/launcher - Runner process exited unexpectedly with error code: 1
```

My tests worked this morning, and I haven't made any changes to the config since then.   The only thing I can think of is Chrome recently updated to v54.  The same failure is happening on multiple machines.  

My config:
```
'use strict';

let HtmlScreenshotReporter = require('protractor-jasmine2-screenshot-reporter');

let reporter = new HtmlScreenshotReporter({
  cleanDestination: true,
  showSummary: true,
  showConfiguration: false,
  reportTitle: null,
  showQuickLinks: true,
  dest: './src/e2e/reports',
  filename: 'report.html',
  reportOnlyFailedSpecs: false,
  captureOnlyFailedSpecs: true,
});

exports.config = {
  baseUrl: 'http://localhost:8080/',

  seleniumAddress: 'http://localhost:4444/wd/hub',

  specs: [
    './tests/**/*.spec.js'
  ],

  exclude: [],

  suites: {
    *suites*
  },

  framework: 'jasmine2',

  allScriptsTimeout: 110000,

  jasmineNodeOpts: {
    showTiming: true,
    showColors: true,
    isVerbose: false,
    includeStackTrace: false,
    defaultTimeoutInterval: 400000
  },

  directConnect: true,

  capabilities: {
    // Browser options: firefox, chrome
    'browserName': 'chrome',
    'platform': 'ANY',
    'version': 'ANY',
    'chromeOptions': {
      // Get rid of --ignore-certificate yellow warning
      args: ['--no-sandbox', '--test-type=browser'],
      // Set download path and avoid prompting for download even though
      // this is already the default on Chrome but for completeness
      prefs: {
        'download': {
          'prompt_for_download': false,
          'default_directory': '//tmp',
        },
      },
    },
    'loggingPrefs': {
      'driver': 'ALL',
      'server': 'ALL',
      'browser': 'ALL'
    },
    shardTestFiles: 'true',
    maxInstances: 10
  },

  beforeLaunch: function() {
    return new Promise(function(resolve){
      reporter.beforeLaunch(resolve);
    });
  },

  onPrepare: function () {
    let SpecReporter = require('jasmine-spec-reporter');
    let JUnitJasmine = require('jasmine-reporters');
    // add jasmine reporters
    jasmine.getEnv().addReporter(new SpecReporter({displayStacktrace: true}));
    jasmine.getEnv().addReporter(reporter);
    jasmine.getEnv().addReporter(
      new JUnitJasmine.JUnitXmlReporter({
        consolidateAll: false,
        savePath: './src/e2e/reports/junit'
      })
    );

    browser.ignoreSynchronization = false;
    require('babel-core/register')({ 'presets': [ 'es2015' ] });
  },

  afterLaunch: function(exitCode) {
    return new Promise(function(resolve){
      reporter.afterLaunch(resolve.bind(this, exitCode));
    });
  },

  /**
   * Angular 2 configuration
   *
   * useAllAngular2AppRoots: tells Protractor to wait for any angular2 apps on the page instead of just the one matching
   * `rootEl`
   *
   */
  useAllAngular2AppRoots: true
};
```

**Comment by nikrolls** [comment]
Happening for us as well, on Codeship using Chrome v53 and Protractor v4.0.9.

```bash
[01:00:58] I/local - Starting selenium standalone server...
[01:00:59] I/local - Selenium standalone server started at http://10.0.3.70:44644/wd/hub

/home/rof/src/github.com/[...]/node_modules/selenium-webdriver/lib/error.js:27
    super(opt_error);
    ^
WebDriverError: Connection refused
    at WebDriverError (/home/rof/src/github.com/[...]/node_modules/selenium-webdriver/lib/error.js:27:5)
    at Object.checkLegacyResponse (/home/rof/src/github.com/[...]/node_modules/selenium-webdriver/lib/error.js:639:15)
    at parseHttpResponse (/home/rof/src/github.com/[...]/node_modules/selenium-webdriver/http/index.js:538:13)
    at client_.send.then.response (/home/rof/src/github.com/[...]/node_modules/selenium-webdriver/http/index.js:472:11)
    at ManagedPromise.invokeCallback_ (/home/rof/src/github.com/[...]/node_modules/selenium-webdriver/lib/promise.js:1379:14)
    at TaskQueue.execute_ (/home/rof/src/github.com/[...]/node_modules/selenium-webdriver/lib/promise.js:2913:14)
    at TaskQueue.executeNext_ (/home/rof/src/github.com/[...]/node_modules/selenium-webdriver/lib/promise.js:2896:21)
    at asyncRun (/home/rof/src/github.com/[...]/node_modules/selenium-webdriver/lib/promise.js:2820:25)
    at /home/rof/src/github.com/[...]/node_modules/selenium-webdriver/lib/promise.js:639:7
    at process._tickCallback (internal/process/next_tick.js:103:7)
From: Task: WebDriver.createSession()
    at Function.createSession (/home/rof/src/github.com/[...]/node_modules/selenium-webdriver/lib/webdriver.js:329:24)
    at Builder.build (/home/rof/src/github.com/[...]/node_modules/selenium-webdriver/builder.js:458:24)
    at Local.DriverProvider.getNewDriver (/home/rof/src/github.com/[...]/node_modules/protractor/built/driverProviders/driverProvider.js:37:33)
    at Runner.createBrowser (/home/rof/src/github.com/[...]/node_modules/protractor/built/runner.js:190:43)
    at /home/rof/src/github.com/[...]/node_modules/protractor/built/runner.js:264:30
    at _fulfilled (/home/rof/src/github.com/[...]/node_modules/q/q.js:834:54)
    at self.promiseDispatch.done (/home/rof/src/github.com/[...]/node_modules/q/q.js:863:30)
    at Promise.promise.promiseDispatch (/home/rof/src/github.com/[...]/node_modules/q/q.js:796:13)
    at /home/rof/src/github.com/[...]/node_modules/q/q.js:604:44
    at runSingle (/home/rof/src/github.com/[...]/node_modules/q/q.js:137:13)
```

**Comment by nikrolls** [comment]
It seems I'm able to resolve it by forcing `webdriver-manager` to a newer version. It was failing using `10.2.3` (installing `chromedriver@2.22`), but when I fiddle with the shrinkwrap and force `10.2.5` (installing `chromedriver@2.24`) everything works.

**Comment by FlorianBoulay** [comment]
Same here, we have upgraded to the last version of protractor and now we have this error : 
```
Message:
    Failed: Error communicating with the remote browser. It may have died.
Build info: version: '2.53.1', revision: 'a36b8b1', time: '2016-06-30 17:37:03'
System info: host: 'railsonfire_7cc5e7b4-4ff0-414a-8e7a-54aa0bba79e1_e451f4323c91', ip: '127.0.1.1', os.name: 'Linux', os.arch: 'amd64', os.version: '4.2.0-42-generic', java.version: '1.7.0_111'
Driver info: driver.version: EventFiringWebDriver
Session ID: 1e2f38bf4c854cb4f28c6225e53f9930
Capabilities [{platform=LINUX, acceptSslCerts=true, javascriptEnabled=true, browserName=chrome, chrome={userDataDir=/tmp/.org.chromium.Chromium.f5utzL, chromedriverVersion=2.24.417424 (c5c5ea873213ee72e3d0929b47482681555340c3)}, networkConnectionEnabled=false, rotatable=false, locationContextEnabled=true, mobileEmulationEnabled=false, version=54.0.2840.59, pageLoadStrategy=normal, takesHeapSnapshot=true, cssSelectorsEnabled=true, databaseEnabled=false, handlesAlerts=true, browserConnectionEnabled=false, webStorageEnabled=true, nativeEvents=true, hasTouchScreen=false, applicationCacheEnabled=false, takesScreenshot=true}]
Stack:
WebDriverError: Error communicating with the remote browser. It may have died.
Build info: version: '2.53.1', revision: 'a36b8b1', time: '2016-06-30 17:37:03'
System info: host: 'railsonfire_7cc5e7b4-4ff0-414a-8e7a-54aa0bba79e1_e451f4323c91', ip: '127.0.1.1', os.name: 'Linux', os.arch: 'amd64', os.version: '4.2.0-42-generic', java.version: '1.7.0_111'
Driver info: driver.version: EventFiringWebDriver
Session ID: 1e2f38bf4c854cb4f28c6225e53f9930
Capabilities [{platform=LINUX, acceptSslCerts=true, javascriptEnabled=true, browserName=chrome, chrome={userDataDir=/tmp/.org.chromium.Chromium.f5utzL, chromedriverVersion=2.24.417424 (c5c5ea873213ee72e3d0929b47482681555340c3)}, networkConnectionEnabled=false, rotatable=false, locationContextEnabled=true, mobileEmulationEnabled=false, version=54.0.2840.59, pageLoadStrategy=normal, takesHeapSnapshot=true, cssSelectorsEnabled=true, databaseEnabled=false, handlesAlerts=true, browserConnectionEnabled=false, webStorageEnabled=true, nativeEvents=true, hasTouchScreen=false, applicationCacheEnabled=false, takesScreenshot=true}]
at WebDriverError (/home/rof/.nvm/versions/node/v4.2.1/lib/node_modules/protractor/node_modules/selenium-webdriver/lib/error.js:27:10)
at Object.checkLegacyResponse (/home/rof/.nvm/versions/node/v4.2.1/lib/node_modules/protractor/node_modules/selenium-webdriver/lib/error.js:639:15)
at parseHttpResponse (/home/rof/.nvm/versions/node/v4.2.1/lib/node_modules/protractor/node_modules/selenium-webdriver/http/index.js:538:13)
at /home/rof/.nvm/versions/node/v4.2.1/lib/node_modules/protractor/node_modules/selenium-webdriver/http/index.js:472:11
at ManagedPromise.invokeCallback_ (/home/rof/.nvm/versions/node/v4.2.1/lib/node_modules/protractor/node_modules/selenium-webdriver/lib/promise.js:1379:14)
at TaskQueue.execute_ (/home/rof/.nvm/versions/node/v4.2.1/lib/node_modules/protractor/node_modules/selenium-webdriver/lib/promise.js:2913:14)
at TaskQueue.executeNext_ (/home/rof/.nvm/versions/node/v4.2.1/lib/node_modules/protractor/node_modules/selenium-webdriver/lib/promise.js:2896:21)
at /home/rof/.nvm/versions/node/v4.2.1/lib/node_modules/protractor/node_modules/selenium-webdriver/lib/promise.js:2820:25
at /home/rof/.nvm/versions/node/v4.2.1/lib/node_modules/protractor/node_modules/selenium-webdriver/lib/promise.js:639:7
at process._tickCallback (node.js:366:9)
From: Task: Protractor.get(http://127.0.0.1:5100/app#/projects/1/show) - resume bootstrap
at WebDriver.schedule (/home/rof/.nvm/versions/node/v4.2.1/lib/node_modules/protractor/node_modules/selenium-webdriver/lib/webdriver.js:377:17)
at ProtractorBrowser.executeScript_ (/home/rof/.nvm/versions/node/v4.2.1/lib/node_modules/protractor/built/browser.js:211:28)
at loadMocks (/home/rof/.nvm/versions/node/v4.2.1/lib/node_modules/protractor/built/browser.js:551:22)
at ManagedPromise.invokeCallback_ (/home/rof/.nvm/versions/node/v4.2.1/lib/node_modules/protractor/node_modules/selenium-webdriver/lib/promise.js:1379:14)
at TaskQueue.execute_ (/home/rof/.nvm/versions/node/v4.2.1/lib/node_modules/protractor/node_modules/selenium-webdriver/lib/promise.js:2913:14)
at TaskQueue.executeNext_ (/home/rof/.nvm/versions/node/v4.2.1/lib/node_modules/protractor/node_modules/selenium-webdriver/lib/promise.js:2896:21)
at /home/rof/.nvm/versions/node/v4.2.1/lib/node_modules/protractor/node_modules/selenium-webdriver/lib/promise.js:2775:27
at /home/rof/.nvm/versions/node/v4.2.1/lib/node_modules/protractor/node_modules/selenium-webdriver/lib/promise.js:639:7
at process._tickCallback (node.js:366:9)
From: Task: Run it("The modal should be hidden") in control flow
at Object.<anonymous> (/home/rof/.nvm/versions/node/v4.2.1/lib/node_modules/protractor/node_modules/jasminewd2/index.js:79:14)
From asynchronous test: 
Error
at Suite.<anonymous> (/home/rof/src/github.com/SparkUpFR/sparkup/spec/javascripts/e2e/tests/app/exit_popup_spec.js:2:3)
at Object.<anonymous> (/home/rof/src/github.com/SparkUpFR/sparkup/spec/javascripts/e2e/tests/app/exit_popup_spec.js:1:1)
at Module._compile (module.js:435:26)
at Object.Module._extensions..js (module.js:442:10)
at Module.load (module.js:356:32)
at Function.Module._load (module.js:311:12)

```
Our cmds : 
rails server -b 127.0.0.1 -p 5100 -d
nvm install 4.2.1
npm install -g protractor
webdriver-manager update
nohup bash -c "webdriver-manager start 2>&1 &" && sleep 10
protractor spec/javascripts/e2e/config/conf.js

**Comment by Xotabu4** [comment]
Seems like same here:

```
[18:13:05] I/direct - Using ChromeDriver directly...                                                                                                         
[18:13:05] I/launcher - Running 1 instances of WebDriver                                                                                                     
[18:13:06] E/launcher - session not created exception                                                                                                        
from unknown error: Runtime.executionContextCreated has invalid 'context': {"auxData":{"frameId":"2189.1","isDefault":true},"id":1,"name":"","origin":"://"} 
  (Session info: chrome=54.0.2840.71)                                                                                                                        
  (Driver info: chromedriver=2.22.397929 (fb72fb249a903a0b1041ea71eb4c8b3fa0d9be5a),platform=Mac OS X 10.11.6 x86_64)                                        
[18:13:06] E/launcher - SessionNotCreatedError: session not created exception                                                                                
from unknown error: Runtime.executionContextCreated has invalid 'context': {"auxData":{"frameId":"2189.1","isDefault":true},"id":1,"name":"","origin":"://"} 
  (Session info: chrome=54.0.2840.71)                                                                                                                        
  (Driver info: chromedriver=2.22.397929 (fb72fb249a903a0b1041ea71eb4c8b3fa0d9be5a),platform=Mac OS X 10.11.6 x86_64)      
```

**Comment by sberan** [comment]
same issue here. `npm install protractor@4.0.8` fixes the issue.

**Comment by mantisbayne** [comment]
Thanks for the replies.  @sberan updating protractor version appears to have fixed the issue for me.  Thanks!

**Comment by sberan** [comment]
Note: this is still an issue in the latest version of protractor. `npm install protractor@4.0.8` is merely a workaround to downgrade to the previous version of protractor. This should not be closed.

**Comment by irushah** [comment]
Is there any workaround for fixing this issue? How did you get the upgraded webdriver-manager?

**Comment by cnishina** [comment]
webdriver-manager was updated yesterday. If you do another `npm install`, npm should fetch the latest webdriver-manager. Another way to do this is to remove your node_modules folder and reinstall your dependencies. This upgrade path is not obvious and we'll probably do a hot fix soon.

**Comment by cnishina** [comment]
@irushah @mantisbayne @Xotabu4 @nikrolls  webdriver-manager was updated yesterday and should download chromedriver 2.24. If you do another `npm install`, npm should fetch the latest webdriver-manager. Another way to do this is to remove your node_modules folder and reinstall your dependencies. This upgrade path is not obvious and we'll probably do a hot fix soon.

@sberan downgrading had some positive effects; however, you probably should still stay on 4.0.9+.  During your downgrade to 4.0.8, npm probably reinstalled webdriver-manager and got the latest.

**Comment by nikrolls** [comment]
Note that if you use shrinkwrap, removing `node_modules` and installing again _won't_ help you. The broken version of `webdriver-manager` still satisfies `protractor`'s version requirement and shrinkwrap will leave it as is. You either need to add `webdriver-manager@10.2.5` as a direct dependency or fiddle with your shrinkwrap to use this version.

**Comment by mantisbayne** [comment]
@cnishina Thanks, everything is working now with the latest version of protractor

**Comment by sberan** [comment]
I can confirm that this is no longer an issue in the latest version of protractor.

**Comment by nikrolls** [comment]
Should we close this issue then?

**Comment by mohitgpt24** [comment]
I am getting the same issue but unable to update the chromedriver to 2.24. I tried deleting the node_modules and then doing npm install, it still shows the version as 2.22. 
I checked the webdriver-manager status, here is the output:

[11:52:29] I/status - selenium standalone version available: 2.53.1 [default]
[11:52:29] I/status - chromedriver version available: 2.22 [default]
[11:52:29] I/status - android-sdk is not present
[11:52:29] I/status - appium is not present

**Comment by FlorianBoulay** [comment]
Not working for us on CodeShip CI.

Our cmds : 
nvm install 4.2.1
npm install -g protractor
npm install -g webdriver-manager@10.2.5
webdriver-manager update
protractor spec/javascripts/e2e/config/conf.js

Output : 
```
Failed: Error communicating with the remote browser. It may have died.
Build info: version: '2.53.1', revision: 'a36b8b1', time: '2016-06-30 17:37:03'
System info: host: 'railsonfire_597ae482-3f38-4101-97f7-5a6f43c709b4_252be9a6ca89', ip: '127.0.1.1', os.name: 'Linux', os.arch: 'amd64', os.version: '4.2.0-42-generic', java.version: '1.7.0_111'
Driver info: driver.version: EventFiringWebDriver
Session ID: 87b47b342c55fbe2a2f3351309dd34bf
Capabilities [{platform=LINUX, acceptSslCerts=true, javascriptEnabled=true, browserName=chrome, chrome={userDataDir=/tmp/.org.chromium.Chromium.2YanSI, chromedriverVersion=2.24.417424 (c5c5ea873213ee72e3d0929b47482681555340c3)}, networkConnectionEnabled=false, rotatable=false, locationContextEnabled=true, mobileEmulationEnabled=false, version=54.0.2840.59, pageLoadStrategy=normal, takesHeapSnapshot=true, cssSelectorsEnabled=true, databaseEnabled=false, handlesAlerts=true, browserConnectionEnabled=false, webStorageEnabled=true, nativeEvents=true, hasTouchScreen=false, applicationCacheEnabled=false, takesScreenshot=true}]
Stack:
WebDriverError: Error communicating with the remote browser. It may have died.
Build info: version: '2.53.1', revision: 'a36b8b1', time: '2016-06-30 17:37:03'
System info: host: 'railsonfire_597ae482-3f38-4101-97f7-5a6f43c709b4_252be9a6ca89', ip: '127.0.1.1', os.name: 'Linux', os.arch: 'amd64', os.version: '4.2.0-42-generic', java.version: '1.7.0_111'
Driver info: driver.version: EventFiringWebDriver
Session ID: 87b47b342c55fbe2a2f3351309dd34bf
Capabilities [{platform=LINUX, acceptSslCerts=true, javascriptEnabled=true, browserName=chrome, chrome={userDataDir=/tmp/.org.chromium.Chromium.2YanSI, chromedriverVersion=2.24.417424 (c5c5ea873213ee72e3d0929b47482681555340c3)}, networkConnectionEnabled=false, rotatable=false, locationContextEnabled=true, mobileEmulationEnabled=false, version=54.0.2840.59, pageLoadStrategy=normal, takesHeapSnapshot=true, cssSelectorsEnabled=true, databaseEnabled=false, handlesAlerts=true, browserConnectionEnabled=false, webStorageEnabled=true, nativeEvents=true, hasTouchScreen=false, applicationCacheEnabled=false, takesScreenshot=true}]
at WebDriverError (/home/rof/.nvm/versions/node/v4.2.1/lib/node_modules/protractor/node_modules/selenium-webdriver/lib/error.js:27:10)
at Object.checkLegacyResponse (/home/rof/.nvm/versions/node/v4.2.1/lib/node_modules/protractor/node_modules/selenium-webdriver/lib/error.js:639:15)
at parseHttpResponse (/home/rof/.nvm/versions/node/v4.2.1/lib/node_modules/protractor/node_modules/selenium-webdriver/http/index.js:538:13)
at /home/rof/.nvm/versions/node/v4.2.1/lib/node_modules/protractor/node_modules/selenium-webdriver/http/index.js:472:11
at ManagedPromise.invokeCallback_ (/home/rof/.nvm/versions/node/v4.2.1/lib/node_modules/protractor/node_modules/selenium-webdriver/lib/promise.js:1379:14)
at TaskQueue.execute_ (/home/rof/.nvm/versions/node/v4.2.1/lib/node_modules/protractor/node_modules/selenium-webdriver/lib/promise.js:2913:14)
at TaskQueue.executeNext_ (/home/rof/.nvm/versions/node/v4.2.1/lib/node_modules/protractor/node_modules/selenium-webdriver/lib/promise.js:2896:21)
at /home/rof/.nvm/versions/node/v4.2.1/lib/node_modules/protractor/node_modules/selenium-webdriver/lib/promise.js:2820:25
at /home/rof/.nvm/versions/node/v4.2.1/lib/node_modules/protractor/node_modules/selenium-webdriver/lib/promise.js:639:7
at process._tickCallback (node.js:366:9)
From: Task: Protractor.get(http://127.0.0.1:5100/app#/projects/1/show) - resume bootstrap
at WebDriver.schedule (/home/rof/.nvm/versions/node/v4.2.1/lib/node_modules/protractor/node_modules/selenium-webdriver/lib/webdriver.js:377:17)
at ProtractorBrowser.executeScript_ (/home/rof/.nvm/versions/node/v4.2.1/lib/node_modules/protractor/built/browser.js:211:28)
at loadMocks (/home/rof/.nvm/versions/node/v4.2.1/lib/node_modules/protractor/built/browser.js:551:22)
at ManagedPromise.invokeCallback_ (/home/rof/.nvm/versions/node/v4.2.1/lib/node_modules/protractor/node_modules/selenium-webdriver/lib/promise.js:1379:14)
at TaskQueue.execute_ (/home/rof/.nvm/versions/node/v4.2.1/lib/node_modules/protractor/node_modules/selenium-webdriver/lib/promise.js:2913:14)
at TaskQueue.executeNext_ (/home/rof/.nvm/versions/node/v4.2.1/lib/node_modules/protractor/node_modules/selenium-webdriver/lib/promise.js:2896:21)
at /home/rof/.nvm/versions/node/v4.2.1/lib/node_modules/protractor/node_modules/selenium-webdriver/lib/promise.js:2775:27
at /home/rof/.nvm/versions/node/v4.2.1/lib/node_modules/protractor/node_modules/selenium-webdriver/lib/promise.js:639:7
at process._tickCallback (node.js:366:9)
From: Task: Run it("The modal should be hidden") in control flow
at Object.<anonymous> (/home/rof/.nvm/versions/node/v4.2.1/lib/node_modules/protractor/node_modules/jasminewd2/index.js:79:14)
From asynchronous test: 
Error
at Suite.<anonymous> (spec/javascripts/e2e/tests/app/exit_popup_spec.js:2:3)
at Object.<anonymous> (spec/javascripts/e2e/tests/app/exit_popup_spec.js:1:1)
at Module._compile (module.js:435:26)
at Object.Module._extensions..js (module.js:442:10)
at Module.load (module.js:356:32)
at Function.Module._load (module.js:311:12)
```

**Comment by nikrolls** [comment]
@mohitgpt24: Which version of `webdriver-manager` is installed? What is the output of `npm list webdriver-manager`?

@FlorianBoulay: What is the output of your `webdriver-manager update` command?

**Comment by FlorianBoulay** [comment]
@nikrolls See bellow : 
```
webdriver-manager: using global installed version 10.2.5
[18:45:31] I/file_manager - creating folder /home/rof/.nvm/versions/node/v4.2.1/lib/node_modules/webdriver-manager/selenium
[18:45:31] I/downloader - selenium standalone: downloading version 2.53.1
[18:45:31] I/downloader - curl -o /home/rof/.nvm/versions/node/v4.2.1/lib/node_modules/webdriver-manager/selenium/selenium-server-standalone-2.53.1.jar https://selenium-release.storage.googleapis.com/2.53/selenium-server-standalone-2.53.1.jar
[18:45:31] I/downloader - chromedriver: downloading version 2.24
[18:45:31] I/downloader - curl -o /home/rof/.nvm/versions/node/v4.2.1/lib/node_modules/webdriver-manager/selenium/chromedriver_2.24linux64.zip https://chromedriver.storage.googleapis.com/2.24/chromedriver_linux64.zip
[18:45:31] I/downloader - geckodriver: downloading version v0.9.0
[18:45:31] I/downloader - curl -o /home/rof/.nvm/versions/node/v4.2.1/lib/node_modules/webdriver-manager/selenium/geckodriver-v0.9.0-linux64.tar.gz https://github.com/mozilla/geckodriver/releases/download/v0.9.0/geckodriver-v0.9.0-linux64.tar.gz
[18:45:32] I/update - chromedriver: unzipping /home/rof/.nvm/versions/node/v4.2.1/lib/node_modules/webdriver-manager/selenium/chromedriver_2.24linux64.zip
[18:45:32] I/update - chromedriver: setting permissions to 0755 for /home/rof/.nvm/versions/node/v4.2.1/lib/node_modules/webdriver-manager/selenium/chromedriver_2.24
[18:45:32] I/update - geckodriver: unzipping /home/rof/.nvm/versions/node/v4.2.1/lib/node_modules/webdriver-manager/selenium/geckodriver-v0.9.0-linux64.tar.gz
[18:45:32] I/update - geckodriver: setting permissions to 0755 for /home/rof/.nvm/versions/node/v4.2.1/lib/node_modules/webdriver-manager/selenium/geckodriver-v0.9.0
```

**Comment by nikrolls** [comment]
@FlorianBoulay: Oh yes, I see it above in the Capabilities line as well. That's interesting, your configuration seems to work for others (myself included). There may be more to the issue.

**Comment by mohitgpt24** [comment]
@nikrolls here is the output of `npm list webdriver-manager` 

`npm ERR! code 1`

**Comment by nikrolls** [comment]
@mohitgpt24: that should not be the case if `webdriver-manager` is installed in the current project.

**Comment by mohitgpt24** [comment]
@nikrolls I am getting `webdriver-manager@10.2.5` and I am getting the same connection refused error when I am executing the tests.

**Comment by nikrolls** [comment]
@mohitgpt24: Looks similar to @FlorianBoulay's issue. `webdriver-manager@10.2.5` / `chromedriver@2.24` seems to fix it for most people, but not all.

**Comment by mohitgpt24** [comment]
What version of protractor is compatible? Should I be using 4.0.9?

**Comment by AlexanderVangelov** [comment]
Hi all, check for console.log() prints inside your code. It is one of the problems after the update.
ref: https://github.com/angular/protractor/issues/3652

**Comment by mohitgpt24** [comment]
I am still unable to solve the issue. Tried installing webdriver-manager and protractor from scratch globally and locally. Here is the output
```
$ protractor --troubleshoot protractor.conf.js
[14:18:46] D/launcher - Running with --troubleshoot
[14:18:46] D/launcher - Protractor version: 4.0.9
[14:18:46] D/launcher - Your base url for tests is undefined
Report destination:   test/e2e-report/screenshots/E2E-report.html
[14:18:46] D/local - Attempting to find the SeleniumServerJar in the default location used by webdriver-manager
[14:18:46] D/local - Attempting to find the chromedriver binary in the default location used by webdriver-manager
[14:18:46] I/local - Starting selenium standalone server...
[14:18:46] I/launcher - Running 1 instances of WebDriver
[14:18:47] I/local - Selenium standalone server started at http://192.168.6.31:54443/wd/hub
[14:18:48] D/runner - WebDriver session successfully started with capabilities Capabilities {
  'applicationCacheEnabled' => false,
  'rotatable' => false,
  'mobileEmulationEnabled' => false,
  'networkConnectionEnabled' => false,
  'chrome' => { chromedriverVersion: '2.24.417412 (ac882d3ce7c0d99292439bf3405780058fcca0a6)',
  userDataDir: '/var/folders/mb/p2lsg5l967jd1ztfqgk5s44c0000gp/T/.org.chromium.Chromium.pnBeBU' },
  'takesHeapSnapshot' => true,
  'pageLoadStrategy' => 'normal',
  'databaseEnabled' => false,
  'handlesAlerts' => true,
  'hasTouchScreen' => false,
  'version' => '54.0.2840.71',
  'platform' => 'MAC',
  'browserConnectionEnabled' => false,
  'nativeEvents' => true,
  'acceptSslCerts' => true,
  'webdriver.remote.sessionid' => 'eae6cfbd-7bbf-485b-826c-844d5071f0d2',
  'locationContextEnabled' => true,
  'webStorageEnabled' => true,
  'browserName' => 'chrome',
  'takesScreenshot' => true,
  'javascriptEnabled' => true,
  'cssSelectorsEnabled' => true }
```

**Comment by mohitgpt24** [comment]
It works for me after downgrading my Google Chrome Version from 54 to 53.

**Comment by mgiambalvo** [comment]
The original issue was using an outdated chromedriver. Running `npm install` to get the latest webdriver-manager and then `webdriver-manager update` to get the latest chromedriver has fixed the original problem.

@mohitgpt24 It looks like you are running chromedriver 2.24, which [should work with Chrome 54](https://sites.google.com/a/chromium.org/chromedriver/downloads). If you're still having problems with Chrome 54, I'd recommend following up on Gitter or StackOverflow, or filing an issue with a repro case. Thanks!

**Comment by FlorianBoulay** [comment]
After removing console.log() that works perfectly for us.

**Comment by jeremypele** [comment]
Works for me when specifying `webdriver-manager@10.2.5` with `chromedriver@2.24`! Thanks @nikrolls :)

**Comment by jyonker** [comment]
Our tests still crashed using `chromedriver@2.24`, but `chromedriver@2.25` seems to have fixed the crash! `webdriver-manager@10.2.5` still thinks that `chromedriver@2.24` is the latest, so you have to force it to install the latest (`webdriver-manager update --versions.chrome 2.25`) and then call it out in your protractor options (`chromeDriver: 'node_modules/protractor/node_modules/webdriver-manager/selenium/chromedriver_2.25'`).
